torch/csrc/jit/codegen/cuda/test/test_gpu.cpp

```diff
@@ -22427,6 +22427,200 @@ TEST_F(NVFuserTest, FusionSerialSmemWriteParallelRead2_CUDA) {
   testValidate(&fusion, cg_outputs, {t0, t1, t2}, {ref}, __LINE__, __FILE__);
 }
 
+// Simple test of async copy primitive
+TEST_F(NVFuserTest, FusionSimpleCpAsync_CUDA) {
+  // requires ampere+ GPU
+  if (!deviceMajorMinorCheck(8)) {
+    GTEST_SKIP() << "skipping tests on pre-AMPERE GPUs";
+    return;
+  }
+
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  int m = 33, n = 31;
+
+  TensorView* tv0 = makeConcreteTensor({m, n});
+  TensorView* tv1 = makeConcreteTensor({m, n});
+
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  TensorView* tv2 = add(tv0, tv1);
+
+  fusion.addOutput(tv2);
+
+  auto tv0_shared = tv0->cacheAfter(LoadStoreOpType::CpAsync);
+  tv0_shared->setMemoryType(MemoryType::Shared);
+
+  tv0->computeAt(tv2, 1);
+  tv0_shared->axis(1)->parallelize(ParallelType::TIDx);
+  tv2->axis(1)->parallelize(ParallelType::TIDx);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({m, n}, options);
+  at::Tensor t1 = at::randn({m, n}, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {t0, t1});
+  auto cg_outputs = fe.runFusion({t0, t1});
+
+  auto ref = t0 + t1;
+
+  testValidate(&fusion, cg_outputs, {t0, t1}, {ref}, __LINE__, __FILE__);
+}
+
+// Simple test of async copy primitive: double buffered
+//   Double buffer case 1, both block sync and async wait
+//  are needed.
+TEST_F(NVFuserTest, FusionDoubleBufferCpAsync1_CUDA) {
+  // requires ampere+ GPU
+  if (!deviceMajorMinorCheck(8)) {
+    GTEST_SKIP() << "skipping tests on pre-AMPERE GPUs";
```

Comment:
I'd prefer these arch-dependent tests go through the lowering and skip the rest so that  they are tested on an older GPU to the extent possible.

Reply:
There's also validation of arch versions at the beginning of lowering, which is inconvenient when I just wanted to see the generated code on my system with a Turing GPU. Perhaps, another `PYTORCH_NVFUSER_DISABLE` flag to skip the validation? 

Certain conditions must hold to make the lowering passes work, but there are also other conditions that do not prevent them to go through, like this use of arch-specific instructions. Another example is non-constant allocations of register arrays. For debugging, I'd like a way to relax those conditions.

Reply:
I think it'd better to have this test run through the scheduling and then do `ASSERT_ANY_THROW` with the lowering on pre Ampere devices.

Reply:
Sure it'd be very helpful debug and test feature. Some extra assertions would be needed in this case to make sure we never send any device mismatched kernel to nvrtc, even nvrtc compile. 

Let's move these discussion and changes into #1702.

Reply:
Are you still working on #1702?

Reply:
Not really. Just rebasing after master bump, and maybe now is almost good time to start merging. 

I will need to propagate changes in #1702 to #1440 , and this one though.